src/components/MessagesList/MessagesGroup/Message/MessagePart/FilePreview.vue

```diff
@@ -472,44 +478,33 @@ export default {
 			event.stopPropagation()
 			event.preventDefault()
 
-			let permissions = ''
-			if (this.permissions) {
-				if (this.permissions & OC.PERMISSION_CREATE) {
-					permissions += 'CK'
-				}
-				if (this.permissions & OC.PERMISSION_READ) {
-					permissions += 'G'
-				}
-				if (this.permissions & OC.PERMISSION_UPDATE) {
-					permissions += 'W'
-				}
-				if (this.permissions & OC.PERMISSION_DELETE) {
-					permissions += 'D'
-				}
-				if (this.permissions & OC.PERMISSION_SHARE) {
-					permissions += 'R'
+			const fileInfo = this.generateViewerObject(this)
+
+			if (this.isSharedItemsTab && this.sharedItemsType === SHARED_ITEM.TYPES.MEDIA) {
+				const token = this.$store.getters.getToken()
```

Comment:
We can add `token` prop to `sharedItems` and pass it to `FilePreview` as we are calling it from the store in both components. All other components that are using `FilePreview` have the token prop already.

Reply:
It may be an overdo + tricky for some components, as they don't have a list, and props are provided not via template:
```js
richParameters[p] = {
	component: FilePreview,
	props: parameters,
}
``` 

But we could do it in a follow-up as well, as other components don't contain multiple media items, and may need this logic it only for NC28

Reply:
Still simple, we can just add token to `parameters`